### DISCUSSION TITLE
Error when creating device with python script - Status Code 500

### DISCUSSION THREAD

**Discussion by julio-oliveira-latitude** [netbox-community/netbox#17652] [Help Wanted!]
I am facing an issue when trying to create devices in NetBox using a Python script. The API call is consistently returning the following error:

`The request failed with code 500 Internal Server Error: {'error': 'Device matching query does not exist.', 'exception': 'DoesNotExist', 'netbox_version': '3.7.8', 'python_version': '3.11.4'}
`
Steps I’ve taken so far:

- I have tried creating the device using curl, the requests library, and the pynetbox library, but the result is the same in all cases.
- I have validated the payload being sent. All values are valid, and all required fields are correctly populated.
- The issue persists despite verifying the correctness of the data being sent to the API.
- I can see that it's possible to create interfaces via API, but the devices creation is continuing to fail.

Here is an example of code that is generating this error:

```
import json
import requests

url_device="[netbox_url]"
token="[my_token]"

payload = {
    "name":"TESTNODE",
    "rack":1,
    "device_type":1,
    "role":1,
    "site":1,
    "status":"planned"
}


def device_creation(payload,token, url_device):
    headers = {
        'accept': 'application/json',
        'Content-Type': 'application/json',
        'Authorization': f'Token {token}'
    }
    response = requests.post(url_device,headers=headers,json=payload)
    if response.status_code == 201:
        print("The device was created")
        device = json.loads(response.text)
        return(json.dumps(device, indent=5))
    else:
        errmessage=(f"Device Creation Failed. Status code: {response.status_code} \n {response.json()}")
        return errmessage

print(device_creation(payload,token,url_device))

```
Can someone help me on this troubleshoot?

Thansks

**Comment by candlerb** [comment]
> url_device="[netbox_url]"

You've hidden the critical information, which is the URL path you're trying to access. It should look something like `http[s]://netbox.example.com/api/dcim/devices/` with the trailing slash.

The following works fine for me with Netbox 4.0.11:

```
export TOKEN="..."
export NETBOX="https://..."
curl -X POST -H "Authorization: Token $TOKEN" -H "Content-Type: application/json" -H  "Accept: application/json; indent=4" "$NETBOX/api/dcim/devices/" --data '{"name":"foobar","role":14,"device_type":1,"status":"active","site":13}'
```
